.grunt-config/webpack.js

```diff
@@ -101,51 +107,127 @@ const baseConfig = {
 	context: __dirname,
 	devtool: 'source-map',
 	externals,
-	plugins,
 	module: moduleRules,
 	resolve: aliasList,
 };
 
-const webpackConfig = Object.assign( {}, baseConfig, {
+const devSharedConfig = {
+	...baseConfig,
+	plugins: [
+		new RemoveChunksPlugin( '.bundle.js' ),
+		...plugins,
+	],
 	mode: 'development',
 	output: {
 		path: path.resolve( __dirname, '../assets/js' ),
+		chunkFilename: '[name].[contenthash].bundle.js',
 		filename: '[name].js',
 		devtoolModuleFilenameTemplate: '../[resource]',
 	},
-	entry: entry,
 	watch: true,
-} );
+};
 
-const webpackProductionConfig = Object.assign( {}, baseConfig, {
+const webpackConfig = [
+	{
+		...devSharedConfig,
+		name: 'base',
+		entry: entry,
+	},
+	{
+		...devSharedConfig,
+		name: 'frontend',
+		optimization: {
+			runtimeChunk:  {
+				name: 'webpack.runtime',
+			},
+			splitChunks: {
+				minChunks: 2,
+			},
+		},
+		entry: frontendEntries,
+	},
+];
+
+const prodSharedOptimization = {
+	minimize: true,
+	minimizer: [
+		new TerserPlugin( {
+			terserOptions: {
+				keep_fnames: true,
+			},
+			include: /\.min\.js$/
+		} ),
+	],
+};
+
+const prodSharedConfig = {
+	...baseConfig,
+	plugins: [
+		new RemoveChunksPlugin( '.bundle.min.js' ),
+		...plugins,
+	],
 	mode: 'production',
 	output: {
 		path: path.resolve( __dirname, '../assets/js' ),
+		chunkFilename: '[name].[contenthash].bundle.min.js',
 		filename: '[name].js',
 	},
-	entry: {},
 	performance: { hints: false },
-	optimization: {
-		minimize: true,
-		minimizer: [
-			new TerserPlugin( {
-				terserOptions: {
-					keep_fnames: true,
-				},
-				include: /\.min\.js$/
-			} ),
-		],
+};
+
+const webpackProductionConfig = [
+	{
+		...prodSharedConfig,
+		name: 'base',
+		entry: {
+			...entry,
+		},
+		optimization: {
+			...prodSharedOptimization,
+		},
 	},
-} );
+	{
+		...prodSharedConfig,
+		name: 'frontend',
+		entry: {
+			...frontendEntries,
+		},
+		optimization: {
+			...prodSharedOptimization,
+			runtimeChunk: {
+				name: 'webpack.runtime.min',
+			},
+			splitChunks: {
+				minChunks: 2,
+			},
+		},
+	},
+];
 
 // Add minified entry points
-for ( const entryPoint in entry ) {
-	webpackProductionConfig.entry[ entryPoint ] = entry[ entryPoint ];
-	webpackProductionConfig.entry[ entryPoint + '.min' ] = entry[ entryPoint ];
-}
+webpackProductionConfig.forEach( ( config, index ) => {
+	for ( const entryPoint in config.entry ) {
+		let entryValue = config.entry[ entryPoint ];
+
+		if ( entryValue.dependOn ) {
+			// We duplicate the 'entryValue' obj for not affecting the 'entry' obj used by the dev process.
+			entryValue = { ...entryValue };
+
+			entryValue.dependOn += '.min';
+		}
+
+		delete config.entry[ entryPoint ];
+		config.entry[ entryPoint + '.min' ] = entryValue;
+	}
+} );
+
+const defaultConfig = webpackConfig.map( ( config ) => {
```

Comment:
It should be the opposite, the default should be `watch: false` and the development should add `watch: true`.

Reply:
The name of the process will be handled on a different PR.